control-plane-operator/hostedclusterconfigoperator/controllers/resources/globalps/pullsecret.go

```diff
@@ -0,0 +1,191 @@
+package globalps
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift/hypershift/support/thirdparty/kubernetes/pkg/credentialprovider"
+	"github.com/openshift/hypershift/support/upsert"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+)
+
+var (
+	maxUnavailable = intstr.FromString("10%")
+	maxSurge       = intstr.FromInt(0)
+)
+
+const (
+	NodePullSecretPath = "/var/lib/kubelet/config.json"
+)
+
+func ReconcileDaemonSet(ctx context.Context, daemonSet *appsv1.DaemonSet, globalPullSecretBytes []byte, c client.Client, createOrUpdate upsert.CreateOrUpdateFN, cpoImage string) error {
+	log, err := logr.FromContext(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get logger: %w", err)
+	}
+
+	log.Info("Reconciling global pull secret daemon set")
+
+	if _, err := createOrUpdate(ctx, c, daemonSet, func() error {
+		daemonSet.Spec = appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": "global-pull-secret-syncer",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"name": "global-pull-secret-syncer",
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName:           "global-pull-secret-syncer",
+					AutomountServiceAccountToken: ptr.To(false),
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsUser: ptr.To[int64](1000),
```

Comment:
I don't think we can't guarantee a particular uid exists in the SCC that will apply to this pod https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/authentication_and_authorization/managing-pod-security-policies#security-context-constraints-example_configuring-internal-oauth